python/pyspark/sql/tests/test_session.py

```diff
@@ -379,6 +381,54 @@ def test_use_custom_class_for_extensions(self):
         )
 
 
+class CreateDataFrame(unittest.TestCase):
```

Comment:
Can we add this test into ArrowTests so it tests Arrow optimization/nonoptimization?

Reply:
Good idea! Adjusted.